### DISCUSSION TITLE
Adjusting timeout for http-extension

### DISCUSSION THREAD

**Discussion by grehnen** [supabase/supabase#13760] [Questions]
Is there a way to adjust the timeout for the http-extension? I've tried using the commands they suggest on their GitHub(https://github.com/pramsey/pgsql-http, I've tried both `http_set_curlopt()` and `http.timeout_msec =`), but nothing seems to work, the timeout is still 5s :(

**Comment by GaryAustin1** [comment]
I've had no luck either in the past.

**Reply by GaryAustin1** [reply]
You might look into the Supabase pg_net extension to see if that will work for you.

**Comment by schopra8** [comment]
I just ran into the same issue --[ apparently the Client object as a timeout parameter that was added a little while back.](https://github.com/supabase-community/postgrest-py/pull/75/commits/2461f07e6b118a42866bc518a0559a10c329b1be)

For example to set it to 180 seconds -- 
```
from supabase import create_client, Client
from supabase.lib.client_options import ClientOptions

supabase: Client = create_client(supabase_url, supabase_key, options=ClientOptions(postgrest_client_timeout=180))

**Reply by 0xemc** [reply]
Lifesaver, thanks @schopra8

**Comment by ghost** [comment]
Try this (to be safe, I would set the timeout within each function block whenever you intend to use the `http` extension):

```plpgsql
CREATE OR REPLACE FUNCTION ...
BEGIN

  -- adjust timeout (in seconds)
  PERFORM http_set_curlopt('CURLOPT_TIMEOUT', '2');
  PERFORM http_set_curlopt('CURLOPT_CONNECTTIMEOUT', '2');
  ...
END;
```

The operative lines are (use `SELECT` instead of `PERFORM` if you are not in a `RETURNS VOID` function):
```plpgsql
PERFORM http_set_curlopt('CURLOPT_TIMEOUT', '2');
PERFORM http_set_curlopt('CURLOPT_CONNECTTIMEOUT', '2');
```
